index.bs

```diff
@@ -1762,8 +1762,9 @@ When this operation is invoked, the authenticator must perform the following pro
 1. Check if at least one of the specified combinations of {{PublicKeyCredentialType}} and cryptographic parameters in
     |credTypesAndPubKeyAlgs| is supported.
     If not, return an error code equivalent to "{{NotSupportedError}}" and terminate the operation.
-1. Check if a credential matching an [=list/item=] of |excludeCredentialDescriptorList| is present on this authenticator. If
-    so, return an error code equivalent to "{{NotAllowedError}}" and terminate the operation.
+1. Check if a credential matching an [=list/item=] of |excludeCredentialDescriptorList| is present on this authenticator for
+    the supplied <code>|rpEntity|.{{PublicKeyCredentialRpEntity/id}}</code>. If so, return an error code equivalent to 
+    "{{NotAllowedError}}" and terminate the operation.
```

Comment:
this step, where we are matching creds bound to an authnr to credDescriptors, is similar to step 16.2 in section [#discover-from-external-source](https://w3c.github.io/webauthn/#discover-from-external-source). Note that the latter step matches on rpId, credential ID, and cred type.  

Should not this authenticatorMakeCredential step similarly match on the same items?

e.g.: 
`1. Check if a credential matching an [=list/item=] of |excludeCredentialDescriptorList| is bound to this authenticator by matching with <code>|rpEntity|.{{PublicKeyCredentialRpEntity/id}}</code>, <code>|excludeCredentialDescriptorList|.{{PublicKeyCredentialDescriptor/id}}</code>, and <code>|excludeCredentialDescriptorList|.{{PublicKeyCredentialDescriptor/type}}</code>. If so, return an error code equivalent to "{{NotAllowedError}}" and terminate the operation.`

Reply:
I believe that is inherent in checking <code>[=list/item]</code> but the check for the rpID is not. It perfectly possible that an item *is* present on the authenticator but for a different rpID while if ID and type are invalid, this cannot be a credential on this authenticator.

Reply:
It seems you @jovasco are saying that we do not need to be explicit regarding the definition of "matching" in this context.  However, the present language in step 16.2 in section [#discover-from-external-source](https://w3c.github.io/webauthn/#discover-from-external-source) is the result of addressing issue #280.  I would not be surprised to have a similar issue raised here down the road if we are not explicit.  Is there any harm/errors with the proposed language above?

Reply:
@equalsJeffH Alright, if we do this, we better do it right:
1. Check if a credential matching an <code>[=list/item=]</code> of <code>|excludeCredentialDescriptorList|</code> is bound to this authenticator with the following procedure:
   1. verify that the <code>|excludeCredentialDescriptorList|.{{PublicKeyCredentialDescriptor/id}}</code> was generated by this Authenticator.
   1.  verify that the <code>|excludeCredentialDescriptorList|.{{PublicKeyCredentialDescriptor/type}}</code> matches the <code>|excludeCredentialDescriptorList|.{{PublicKeyCredentialDescriptor/id}}</code> 
   1. verify that <code>|rpEntity|.{{PublicKeyCredentialRpEntity/id}}</code> is identical to the one used to create the <code>|excludeCredentialDescriptorList|.{{PublicKeyCredentialDescriptor/id}}</code>.

If all of these check succeed, return an error code equivalent to "{{NotAllowedError}}" and terminate the operation.

If you agree I will update the pull request for both make crednetial and getassertion.

Reply:
@jovasco proposed:
> 1. Check if a credential matching an <code>[=list/item=]</code> of <code>|excludeCredentialDescriptorList|</code> is bound to this authenticator with the following procedure:

hm, I think i see whay you are getting at -- i.e., being even more explicit -- though, this does not seem correct to me...in the above, do you mean to also say "...for each item of |excludeCredentialDescriptorList| ?

>          i. verify that the <code>|excludeCredentialDescriptorList|.
>              {{PublicKeyCredentialDescriptor/id}}</code> 
>              was generated by this Authenticator.

in the above, "generated by this authnr" is not what we are looking for. Rather, we are trying to ascertain whether there exists a cred bound to this authnr having a credential ID matching `|excludeCredentialDescriptorList|.{{PublicKeyCredentialDescriptor/id}}`.

>         ii.  verify that the <code>|excludeCredentialDescriptorList|.
>              {{PublicKeyCredentialDescriptor/type}}</code> matches the
>             <code>|excludeCredentialDescriptorList|.{{PublicKeyCredentialDescriptor/id}}</code> 

in the above, do you mean to say something like "if there exists a cred bound to this authnr whose `{{PublicKeyCredentialDescriptor/type}}` value matches `|excludeCredentialDescriptorList|.{{PublicKeyCredentialDescriptor/type}}` ?

>         iii. verify that <code>|rpEntity|.{{PublicKeyCredentialRpEntity/id}}</code> is 
>              identical to the one used to create the <code>|excludeCredentialDescriptorList|.
>              {{PublicKeyCredentialDescriptor/id}}</code>.

in the above, do you mean to say something like "if there exists a cred bound to this authnr whose RP ID matches `|excludeCredentialDescriptorList|.{{PublicKeyCredentialDescriptor/id}}`" ?


Perhaps the below works ?

`3. Check if any credential, bound to this authenticator and matching an 
[=list/item=] of |excludeCredentialDescriptorList|, exists.  
    A match occurs if a credential matches all of an 
|excludeCredentialDescriptorList| item's <code>|rpEntity|.
{{PublicKeyCredentialRpEntity/id}}</code>, <code>|excludeCredentialDescriptorList|.
{{PublicKeyCredentialDescriptor/id}}</code>, and 
<code>|excludeCredentialDescriptorList|.{{PublicKeyCredentialDescriptor/type}}</code>.
     If so, return an error code equivalent to "{{NotAllowedError}}" and 
terminate the operation.`

Reply:
@equalsJeffH  I modified it slightly. rpId is not part of the exclude list item and I found the language a bit tortured.